ta/os_test/cxx_tests.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019 Huawei Technologies Co., Ltd
+ */
+
+
+extern "C" {
+
+#include <dlfcn.h>
+#include <stdint.h>
+#include <tee_ta_api.h>
+#include <os_test_lib.h>
+
+#include "os_test.h"
+
+};
+
+class CtorTest {
+public:
+	CtorTest() : val(1) {}
+
+	int val;
+};
+
+static CtorTest ctor_test;
+
+TEE_Result ta_entry_cxx_ctor_main(void)
+{
+	if (ctor_test.val != 1)
+		return TEE_ERROR_GENERIC;
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result ta_entry_cxx_ctor_shlib(void)
+{
+	return os_test_shlib_cxx_ctor();
+}
+
+TEE_Result ta_entry_cxx_ctor_shlib_dl(void)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_Result (*ctor_test_fn)(void);
+	void *handle = NULL;
+
+	handle = dlopen("b3091a65-9751-4784-abf7-0298a7cc35ba",
+			RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
+	if (!handle)
+		return TEE_ERROR_GENERIC;
+
+	ctor_test_fn = (TEE_Result (*)(void))dlsym(handle,
+						   "os_test_shlib_dl_cxx_ctor");
+	if (!ctor_test_fn)
+		goto err;
+	res = ctor_test_fn();
+err:
```

Comment:
maybe:
```c
	if (ctor_test_fn)
		res = ctor_test_fn();
```
unless you expect more tests here?

Reply:
No more tests for now so I'll write as you suggest